docs/source/data-structures.rst

```diff
@@ -0,0 +1,173 @@
+- ``parent``: The single ``DataTree`` object whose children this datatree is a member of, known as its' "parent node".
+
+Each child automatically knows about its parent node, and a node without a parent is known as a "root" node
+(represented by the ``parent`` attribute pointing to ``None``).
+Nodes can have multiple children, but as each child node has at most one parent, there can only ever be one root node in a given tree.
+
+The overall structure is technically a `connected acyclic undirected rooted graph`, otherwise known as a
+`"Tree" <https://en.wikipedia.org/wiki/Tree_(graph_theory)>`_.
+
+.. note::
+
+    Technically a ``DataTree`` with more than one child node forms an `"Ordered Tree" <https://en.wikipedia.org/wiki/Tree_(graph_theory)#Ordered_tree>`_,
+    because the children are stored in an Ordered Dictionary. However, this distinction only really matters for a few
+    edge cases involving operations on multiple trees simultaneously, and can safely be ignored by most users.
+
+
+``DataTree`` objects can also optionally have a ``name`` as well as ``attrs``, just like a ``DataArray``.
+Again these are not normally used unless explicitly accessed by the user.
+
+
+Creating a DataTree
+~~~~~~~~~~~~~~~~~~~
+
+There are two ways to create a ``DataTree`` from scratch. The first is to create each node individually,
+specifying the nodes' relationship to one another as you create each one.
+
+The ``DataTree`` constructor takes:
+
+- ``data``: The data that will be stored in this node, represented by a single ``xarray.Dataset``, or a named ``xarray.DataArray``.
+- ``parent``: The parent node (if there is one), given as a ``DataTree`` object.
+- ``children``: The various child nodes (if there are any), given as a mapping from string keys to ``DataTree`` objects.
+- ``name``: A string to use as the name of this node.
+
+Let's make a datatree node without anything in it:
+
+.. ipython:: python
+
+    from datatree import DataTree
+
+    # create root node
+    node1 = DataTree(name="Oak")
+
+    node1
+
+At this point our node is also the root node, as every tree has a root node.
+
+We can add a second node to this tree either by referring to the first node in the constructor of the second:
+
+.. ipython:: python
+
+    # add a child by referring to the parent node
+    node2 = DataTree(name="Bonsai", parent=node1)
+
+or by dynamically updating the attributes of one node to refer to another:
+
+.. ipython:: python
+
+    # add a grandparent by updating the .parent property of an existing node
+    node0 = DataTree(name="General Sherman")
+    node1.parent = node0
+
+Our tree now has three nodes within it, and one of the two new nodes has become the new root:
+
+.. ipython:: python
+
+    node0
+
+Is is at tree construction time that consistency checks are enforced. For instance, if we try to create a `cycle` the constructor will raise an error:
+
+.. ipython:: python
+    :okexcept:
+
+    node0.parent = node2
```

Comment:
This should raise an error but with #41 it currently doesn't - I must remember to fix that, as it's surprising that it's not tested.

Reply:
Should be fixed by #105